### PR TITLE
LPS-186037 Automate Liferay.FeatureFlags

### PR DESCRIPTION
**Pull request by eltonccdantas**
This pull request aims at the automation of tests considered critical contained in `FeatureFlags.testcase`

- [x] [FeatureFlags#CanBeEnabled]
- [x] [FeatureFlags#CanBeDisabled]

[Jira Ticket](https://issues.liferay.com/browse/LPS-186037)

### CODE REVIEW

File: modules/apps/feature-flag/feature-flag-web/src/FeatureFlags.testcase [hunk 1]
```diff
@@ -0,0 +1,79 @@
+@component-name = "feature flags"
```
Comment (john-co) [inline modules/apps/feature-flag/feature-flag-web/src/FeatureFlags.testcase:1]: John can update this line and test.properties component.names list with `portal-feature-flags`

File: modules/apps/feature-flag/feature-flag-web/src/FeatureFlags.testcase [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+	tearDown {
+		var testPortalInstance = PropsUtil.get("test.portal.instance");
+
+		if (${testPortalInstance} == "true") {
+			PortalInstances.tearDownCP();
+		}
+	}
+
+	@description = "LPS-186037 - This tests asserts that a feature flag can be Disabled"
```
Comment (john-co) [inline modules/apps/feature-flag/feature-flag-web/src/FeatureFlags.testcase:31]: update to story ticket: LPS-167698

File: modules/apps/feature-flag/feature-flag-web/src/FeatureFlags.testcase [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+		task ("And a feature flag is enabled") {
+			Click(
+				key_labelName = "Disabled",
+				locator1 = "CalendarLookAndFeelConfiguration#TOGGLE_SWITCH_LABEL");
+		}
+
+		task ("When the administrator disables a feature flag") {
+			Click(
+				key_labelName = "Enabled",
+				locator1 = "CalendarLookAndFeelConfiguration#TOGGLE_SWITCH_LABEL");
```
Comment (john-co) [inline modules/apps/feature-flag/feature-flag-web/src/FeatureFlags.testcase:45]: If this path is the only path that works for our usecase in Feature Flags, then make a copy of this and create a new path file atleast in `modules/apps/feature-flag/feature-flag-test/src/testFunctional/paths/FeatureFlags.path`

File: modules/apps/feature-flag/feature-flag-web/src/FeatureFlags.testcase [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+		}
+	}
+
+	@description = "LPS-186037 - This tests asserts that a feature flag can be Disabled"
+	@priority = 5
+	test CanBeDisabled {
+		property portal.acceptance = "true";
+
+		task ("And a feature flag is enabled") {
+			Click(
```
Comment (john-co) [inline modules/apps/feature-flag/feature-flag-web/src/FeatureFlags.testcase:37]: Try having a path like `//*[contains(text(),'${name}')]//ancestor::dd//label//input[contains(@class,'toggle-switch-check')]` and using function `Click.toggleSwitch`

File: modules/apps/feature-flag/feature-flag-web/src/FeatureFlags.testcase [hunk 5]
```diff
@@ -0,0 +1,79 @@
+@component-name = "feature flags"
+definition {
+
+	property portal.acceptance = "true";
```
Comment (john-co) [inline modules/apps/feature-flag/feature-flag-web/src/FeatureFlags.testcase:4]: Remember to add property custom.properties:

feature.flag.LPS-xxxxxxx=false
feature.flag.LPS-xxxxxx.status=beta
feature.flag.LPS-xxxxxx.title=Terms and Conditions pages
feature.flag.LPS-xxxxxx.description=Create your own terms and conditions pages

Comment (john-co) [reply modules/apps/feature-flag/feature-flag-web/src/FeatureFlags.testcase:4]: feature.flag.LPS-yyyyyy=false
feature.flag.LPS-yyyyyy.status=beta
feature.flag.LPS-yyyyyy.dependencies=LPS-xxxxxx

File: modules/apps/feature-flag/feature-flag-web/src/FeatureFlags.testcase [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+		}
+	}
+
+	@description = "LPS-186037 - This tests asserts that a feature flag can be enabled"
+	@priority = 5
+	test CanBeEnabled {
+		property portal.acceptance = "true";
+
+		task ("And a feature flag is disabled") {
+			AssertElementPresent(
```
Comment (john-co) [inline modules/apps/feature-flag/feature-flag-web/src/FeatureFlags.testcase:61]: I suggest creating an if else statement to toggle to disabled if the feature flag is enabled. 

And then use VerifyElementPresent on a path something like this:

`//*[contains(text(),'${title-row-name}')]//ancestor::dd//label//span[contains(@class,'toggle-switch-label')]`

File: portal-web/poshi.properties [hunk 7]
```diff
@@ -16,6 +16,7 @@ test.dirs=\
     ../modules/apps/digital-signature/digital-signature-test/src/testFunctional,\
     ../modules/apps/dispatch/dispatch-test/src/testFunctional,\
     ../modules/apps/document-library-opener/document-library-opener-google-drive-test/src/testFunctional,\
+    ../modules/apps/feature-flag/feature-flag-web/src/,\
```
Comment (john-co) [inline portal-web/poshi.properties:19]: update to new directory

**Comment by eltonccdantas** [comment]
Thanks for reviewing it... I intend to refactor it Tomorrow. 😄  👨🏻‍💻

File: modules/apps/feature-flag/feature-flag-web/src/FeatureFlags.testcase (hunk 1 above)
[the remark this answers is not in the document]
Comment (eltonccdantas) [reply modules/apps/feature-flag/feature-flag-web/src/FeatureFlags.testcase]: done

File: modules/apps/feature-flag/feature-flag-web/src/FeatureFlags.testcase (hunk 3 above)
Comment (eltonccdantas) [reply modules/apps/feature-flag/feature-flag-web/src/FeatureFlags.testcase]: done

**Comment by eltonccdantas** [comment]
ci:test:sf

**Comment by eltonccdantas** [comment]
ci:test:sf

**Comment by eltonccdantas** [comment]
ci:test:peds-sm

**Comment by john-co** [comment]
ci:test:peds-sm

File: modules/apps/feature-flag/feature-flag-test/src/testFunctional/tests/FeatureFlags.testcase [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+		}
+	}
+
+	@description = "LPS-167698 - This tests asserts that a feature flag can be Disabled"
+	@priority = 5
+	test CanBeDisabled {
+		property portal.acceptance = "true";
+
+		task ("And a feature flag is enabled") {
+			if (IsElementPresent(locator1 = "FeatureFlags#TOGGLE_ROW", title_row = "Proxy Object", toggle_state = "Disabled")) {
```
Comment (john-co) [inline modules/apps/feature-flag/feature-flag-test/src/testFunctional/tests/FeatureFlags.testcase:38]: please target the sample feature flag used in custom.properties.

**Review by john-co — CHANGES_REQUESTED** [review]
Only one comment, otherwise looks good. Let's see what Drew says on troubleshooting the portal-ext.properties.

File: modules/apps/feature-flag/feature-flag-test/src/testFunctional/tests/FeatureFlags.testcase [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+
+		task ("When the administrator enables a feature flag") {
+			Check.checkToggleSwitch(
+				locator1 = "FeatureFlags#TOGGLE_ROW",
+				title_row = "Terms and Conditions pages",
+				toggle_state = "Disabled");
+		}
+
+		task ("Then assert the toggle is enabled state") {
+			VerifyElementPresent(
```
Comment (john-co) [inline modules/apps/feature-flag/feature-flag-test/src/testFunctional/tests/FeatureFlags.testcase:78]: change to AssertElementPresent

File: modules/apps/feature-flag/feature-flag-test/src/testFunctional/tests/FeatureFlags.testcase [hunk 10]
```diff
@@ -0,0 +1,85 @@
+@component-name = "portal-feature-flags"
+definition {
+
+	property custom.properties = "feature.flag.LPS-xxxxxx=false${line.separator}feature.flag.LPS-xxxxxx.status=beta${line.separator}feature.flag.LPS-xxxxxx.title=Terms and Conditions pages${line.separator}feature.flag.LPS-xxxxxx.description=Create your own terms and conditions pages${line.separator}feature.flag.LPS-yyyyyy=false${line.separator}feature.flag.LPS-yyyyyy.status=beta${line.separator}feature.flag.LPS-yyyyyy.dependencies=LPS-xxxxxx";
```
Comment (john-co) [inline modules/apps/feature-flag/feature-flag-test/src/testFunctional/tests/FeatureFlags.testcase:4]: feature.flag.LPS-yyyyyy.status change to feature.flag.LPS-yyyyyy.type, also feature.flag.LPS-xxxxxx.type

File: modules/apps/feature-flag/feature-flag-test/src/testFunctional/tests/FeatureFlags.testcase [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+
+		task ("When the administrator disables a feature flag") {
+			Check.checkToggleSwitch(
+				locator1 = "FeatureFlags#TOGGLE_ROW",
+				title_row = "Terms and Conditions pages",
+				toggle_state = "Enabled");
+		}
+
+		task ("Then assert the toggle is disabled state") {
+			VerifyElementPresent(
```
Comment (john-co) [inline modules/apps/feature-flag/feature-flag-test/src/testFunctional/tests/FeatureFlags.testcase:51]: AssertElementPresent

Comment (eltonccdantas) [reply modules/apps/feature-flag/feature-flag-test/src/testFunctional/tests/FeatureFlags.testcase]: Done

File: modules/apps/feature-flag/feature-flag-test/src/testFunctional/tests/FeatureFlags.testcase (hunk 10 above)
Comment (eltonccdantas) [reply modules/apps/feature-flag/feature-flag-test/src/testFunctional/tests/FeatureFlags.testcase]: Done

File: modules/apps/feature-flag/feature-flag-test/src/testFunctional/tests/FeatureFlags.testcase (hunk 9 above)
Comment (eltonccdantas) [reply modules/apps/feature-flag/feature-flag-test/src/testFunctional/tests/FeatureFlags.testcase]: Done

**Comment by eltonccdantas** [comment]
ci:test:sf

**Comment by eltonccdantas** [comment]
ci:test:peds-sm

**Comment by eltonccdantas** [comment]
ci:test:relevant

**Comment by eltonccdantas** [comment]
ci:forward

**Comment by eltonccdantas** [comment]
ci:forward

**Pull request closed without merging**